src/ScalarAuthClient.js

```diff
@@ -55,23 +59,11 @@ class ScalarAuthClient {
         if (!token) {
             return this.registerForToken();
         } else {
-            return this.validateToken(token).then(userId => {
-                const me = MatrixClientPeg.get().getUserId();
-                if (userId !== me) {
-                    throw new Error("Scalar token is owned by someone else: " + me);
-                }
-                return token;
-            }).catch(err => {
-                console.error(err);
-
-                // Something went wrong - try to get a new token.
```

Comment:
This step has gone missing, which leads to complications when the token is deleted from under the user.

Reply:
good catch - should be fixed now